From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Mon, 11 Dec 2017 22:42:11 +0100
Subject: Add English-only search engine

Add a Google search engine that forces languages to English,
disable from all its searches RLZ and field experiments querystring parameters.

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 .../search_engines/prepopulated_engines.json  |  21 ++
 .../search_engines/search_engine_type.h       |   1 +
 .../template_url_prepopulate_data.cc          | 216 +++++++++---------
 3 files changed, 130 insertions(+), 108 deletions(-)

diff --git a/components/search_engines/prepopulated_engines.json b/components/search_engines/prepopulated_engines.json
--- a/components/search_engines/prepopulated_engines.json
+++ b/components/search_engines/prepopulated_engines.json
@@ -137,6 +137,27 @@
       "id": 1
     },
 
+    "googleen": {
+      "name": "Google in English",
+      "keyword": "googleen",
+      "favicon_url": "https://www.google.com/favicon.ico",
+      "search_url": "{google:baseURL}search?q={searchTerms}&ie={inputEncoding}&hl=en",
+      "suggest_url": "{google:baseSuggestURL}search?client={google:suggestClient}&q={searchTerms}&hl=en",
+      "image_url": "{google:baseURL}searchbyimage/upload?hl=en",
+      "new_tab_url": "{google:baseURL}_/chrome/newtab?hl=en&ie={inputEncoding}",
+      "contextual_search_url": "{google:baseURL}_/contextualsearch?{google:contextualSearchVersion}{google:contextualSearchContextData}&hl=en",
+      "image_url_post_params": "encoded_image={google:imageThumbnail},image_url={google:imageURL},sbisrc={google:imageSearchSource},original_width={google:imageOriginalWidth},original_height={google:imageOriginalHeight}",
+      "alternate_urls": [
+        "{google:baseURL}?hl=en#q={searchTerms}",
+        "{google:baseURL}search?hl=en#q={searchTerms}",
+        "{google:baseURL}webhp?hl=en#q={searchTerms}",
+        "{google:baseURL}s?hl=en#q={searchTerms}",
+        "{google:baseURL}s?hl=en&q={searchTerms}"
+      ],
+      "type": "SEARCH_ENGINE_GOOGLE_EN",
+      "id": 13
+    },
+
     "mail_ru": {
       "name": "@MAIL.RU",
       "keyword": "mail.ru",
diff --git a/components/search_engines/search_engine_type.h b/components/search_engines/search_engine_type.h
--- a/components/search_engines/search_engine_type.h
+++ b/components/search_engines/search_engine_type.h
@@ -78,6 +78,7 @@ enum SearchEngineType {
   SEARCH_ENGINE_STARTER_PACK_BOOKMARKS = 61,
   SEARCH_ENGINE_STARTER_PACK_HISTORY = 62,
   SEARCH_ENGINE_STARTER_PACK_TABS = 63,
+  SEARCH_ENGINE_GOOGLE_EN = 64,
 
   SEARCH_ENGINE_MAX  // Bounding value needed for UMA histogram macro.
 };
diff --git a/components/search_engines/template_url_prepopulate_data.cc b/components/search_engines/template_url_prepopulate_data.cc
--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -29,7 +29,7 @@ namespace {
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* const engines_default[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
 };
@@ -39,7 +39,7 @@ const PrepopulatedEngine* const engines_default[] = {
 // clang-format off
 // United Arab Emirates
 const PrepopulatedEngine* const engines_AE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -48,7 +48,7 @@ const PrepopulatedEngine* const engines_AE[] = {
 
 // Albania
 const PrepopulatedEngine* const engines_AL[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -57,7 +57,7 @@ const PrepopulatedEngine* const engines_AL[] = {
 
 // Argentina
 const PrepopulatedEngine* const engines_AR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_ar,
     &duckduckgo,
@@ -66,7 +66,7 @@ const PrepopulatedEngine* const engines_AR[] = {
 
 // Austria
 const PrepopulatedEngine* const engines_AT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo_at,
@@ -75,7 +75,7 @@ const PrepopulatedEngine* const engines_AT[] = {
 
 // Australia
 const PrepopulatedEngine* const engines_AU[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_au,
     &duckduckgo,
@@ -84,7 +84,7 @@ const PrepopulatedEngine* const engines_AU[] = {
 
 // Bosnia and Herzegovina
 const PrepopulatedEngine* const engines_BA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -93,7 +93,7 @@ const PrepopulatedEngine* const engines_BA[] = {
 
 // Belgium
 const PrepopulatedEngine* const engines_BE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &ecosia,
@@ -102,7 +102,7 @@ const PrepopulatedEngine* const engines_BE[] = {
 
 // Bulgaria
 const PrepopulatedEngine* const engines_BG[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -111,7 +111,7 @@ const PrepopulatedEngine* const engines_BG[] = {
 
 // Bahrain
 const PrepopulatedEngine* const engines_BH[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -120,7 +120,7 @@ const PrepopulatedEngine* const engines_BH[] = {
 
 // Burundi
 const PrepopulatedEngine* const engines_BI[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -129,7 +129,7 @@ const PrepopulatedEngine* const engines_BI[] = {
 
 // Brunei
 const PrepopulatedEngine* const engines_BN[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -138,7 +138,7 @@ const PrepopulatedEngine* const engines_BN[] = {
 
 // Bolivia
 const PrepopulatedEngine* const engines_BO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -147,7 +147,7 @@ const PrepopulatedEngine* const engines_BO[] = {
 
 // Brazil
 const PrepopulatedEngine* const engines_BR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_br,
     &duckduckgo,
@@ -156,7 +156,7 @@ const PrepopulatedEngine* const engines_BR[] = {
 
 // Belarus
 const PrepopulatedEngine* const engines_BY[] = {
-    &google,
+    &googleen, &google,
     &yandex_by,
     &mail_ru,
     &bing,
@@ -165,7 +165,7 @@ const PrepopulatedEngine* const engines_BY[] = {
 
 // Belize
 const PrepopulatedEngine* const engines_BZ[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -174,7 +174,7 @@ const PrepopulatedEngine* const engines_BZ[] = {
 
 // Canada
 const PrepopulatedEngine* const engines_CA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_ca,
     &duckduckgo,
@@ -183,7 +183,7 @@ const PrepopulatedEngine* const engines_CA[] = {
 
 // Switzerland
 const PrepopulatedEngine* const engines_CH[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &ecosia,
@@ -192,7 +192,7 @@ const PrepopulatedEngine* const engines_CH[] = {
 
 // Chile
 const PrepopulatedEngine* const engines_CL[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_cl,
     &duckduckgo,
@@ -204,13 +204,13 @@ const PrepopulatedEngine* const engines_CN[] = {
     &sogou,
     &baidu,
     &bing,
-    &google,
+    &googleen, &google,
     &so_360,
 };
 
 // Colombia
 const PrepopulatedEngine* const engines_CO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_co,
     &ecosia,
@@ -219,7 +219,7 @@ const PrepopulatedEngine* const engines_CO[] = {
 
 // Costa Rica
 const PrepopulatedEngine* const engines_CR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -228,7 +228,7 @@ const PrepopulatedEngine* const engines_CR[] = {
 
 // Czech Republic
 const PrepopulatedEngine* const engines_CZ[] = {
-    &google,
+    &googleen, &google,
     &seznam_cz,
     &bing,
     &yahoo,
@@ -237,7 +237,7 @@ const PrepopulatedEngine* const engines_CZ[] = {
 
 // Germany
 const PrepopulatedEngine* const engines_DE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &ecosia,
     &duckduckgo,
@@ -246,7 +246,7 @@ const PrepopulatedEngine* const engines_DE[] = {
 
 // Denmark
 const PrepopulatedEngine* const engines_DK[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_dk,
     &duckduckgo,
@@ -255,7 +255,7 @@ const PrepopulatedEngine* const engines_DK[] = {
 
 // Dominican Republic
 const PrepopulatedEngine* const engines_DO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -264,7 +264,7 @@ const PrepopulatedEngine* const engines_DO[] = {
 
 // Algeria
 const PrepopulatedEngine* const engines_DZ[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_fr,
     &yandex_com,
@@ -273,7 +273,7 @@ const PrepopulatedEngine* const engines_DZ[] = {
 
 // Ecuador
 const PrepopulatedEngine* const engines_EC[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &ecosia,
@@ -282,7 +282,7 @@ const PrepopulatedEngine* const engines_EC[] = {
 
 // Estonia
 const PrepopulatedEngine* const engines_EE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yandex_ru,
     &duckduckgo,
@@ -291,7 +291,7 @@ const PrepopulatedEngine* const engines_EE[] = {
 
 // Egypt
 const PrepopulatedEngine* const engines_EG[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_com,
@@ -300,7 +300,7 @@ const PrepopulatedEngine* const engines_EG[] = {
 
 // Spain
 const PrepopulatedEngine* const engines_ES[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -309,7 +309,7 @@ const PrepopulatedEngine* const engines_ES[] = {
 
 // Finland
 const PrepopulatedEngine* const engines_FI[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_fi,
     &duckduckgo,
@@ -318,7 +318,7 @@ const PrepopulatedEngine* const engines_FI[] = {
 
 // Faroe Islands
 const PrepopulatedEngine* const engines_FO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_dk,
     &duckduckgo,
@@ -327,7 +327,7 @@ const PrepopulatedEngine* const engines_FO[] = {
 
 // France
 const PrepopulatedEngine* const engines_FR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_fr,
     &qwant,
@@ -336,7 +336,7 @@ const PrepopulatedEngine* const engines_FR[] = {
 
 // United Kingdom
 const PrepopulatedEngine* const engines_GB[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_uk,
     &duckduckgo,
@@ -345,7 +345,7 @@ const PrepopulatedEngine* const engines_GB[] = {
 
 // Greece
 const PrepopulatedEngine* const engines_GR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -354,7 +354,7 @@ const PrepopulatedEngine* const engines_GR[] = {
 
 // Guatemala
 const PrepopulatedEngine* const engines_GT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -363,7 +363,7 @@ const PrepopulatedEngine* const engines_GT[] = {
 
 // Hong Kong
 const PrepopulatedEngine* const engines_HK[] = {
-    &google,
+    &googleen, &google,
     &yahoo_hk,
     &bing,
     &baidu,
@@ -372,7 +372,7 @@ const PrepopulatedEngine* const engines_HK[] = {
 
 // Honduras
 const PrepopulatedEngine* const engines_HN[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -381,7 +381,7 @@ const PrepopulatedEngine* const engines_HN[] = {
 
 // Croatia
 const PrepopulatedEngine* const engines_HR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -390,7 +390,7 @@ const PrepopulatedEngine* const engines_HR[] = {
 
 // Hungary
 const PrepopulatedEngine* const engines_HU[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -399,7 +399,7 @@ const PrepopulatedEngine* const engines_HU[] = {
 
 // Indonesia
 const PrepopulatedEngine* const engines_ID[] = {
-    &google,
+    &googleen, &google,
     &yahoo_id,
     &bing,
     &yandex_com,
@@ -408,7 +408,7 @@ const PrepopulatedEngine* const engines_ID[] = {
 
 // Ireland
 const PrepopulatedEngine* const engines_IE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_uk,
     &duckduckgo,
@@ -417,7 +417,7 @@ const PrepopulatedEngine* const engines_IE[] = {
 
 // Israel
 const PrepopulatedEngine* const engines_IL[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yandex_ru,
     &yahoo,
@@ -426,7 +426,7 @@ const PrepopulatedEngine* const engines_IL[] = {
 
 // India
 const PrepopulatedEngine* const engines_IN[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_in,
     &duckduckgo,
@@ -435,7 +435,7 @@ const PrepopulatedEngine* const engines_IN[] = {
 
 // Iraq
 const PrepopulatedEngine* const engines_IQ[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_tr,
@@ -444,7 +444,7 @@ const PrepopulatedEngine* const engines_IQ[] = {
 
 // Iran
 const PrepopulatedEngine* const engines_IR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &ask,
@@ -453,7 +453,7 @@ const PrepopulatedEngine* const engines_IR[] = {
 
 // Iceland
 const PrepopulatedEngine* const engines_IS[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo,
@@ -462,7 +462,7 @@ const PrepopulatedEngine* const engines_IS[] = {
 
 // Italy
 const PrepopulatedEngine* const engines_IT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -471,7 +471,7 @@ const PrepopulatedEngine* const engines_IT[] = {
 
 // Jamaica
 const PrepopulatedEngine* const engines_JM[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -480,7 +480,7 @@ const PrepopulatedEngine* const engines_JM[] = {
 
 // Jordan
 const PrepopulatedEngine* const engines_JO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -489,7 +489,7 @@ const PrepopulatedEngine* const engines_JO[] = {
 
 // Japan
 const PrepopulatedEngine* const engines_JP[] = {
-    &google,
+    &googleen, &google,
     &yahoo_jp,
     &bing,
     &duckduckgo,
@@ -498,7 +498,7 @@ const PrepopulatedEngine* const engines_JP[] = {
 
 // Kenya
 const PrepopulatedEngine* const engines_KE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -507,7 +507,7 @@ const PrepopulatedEngine* const engines_KE[] = {
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
-    &google,
+    &googleen, &google,
     &naver,
     &daum,
     &bing,
@@ -516,7 +516,7 @@ const PrepopulatedEngine* const engines_KR[] = {
 
 // Kuwait
 const PrepopulatedEngine* const engines_KW[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -525,7 +525,7 @@ const PrepopulatedEngine* const engines_KW[] = {
 
 // Kazakhstan
 const PrepopulatedEngine* const engines_KZ[] = {
-    &google,
+    &googleen, &google,
     &yandex_kz,
     &mail_ru,
     &bing,
@@ -534,7 +534,7 @@ const PrepopulatedEngine* const engines_KZ[] = {
 
 // Lebanon
 const PrepopulatedEngine* const engines_LB[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -543,7 +543,7 @@ const PrepopulatedEngine* const engines_LB[] = {
 
 // Liechtenstein
 const PrepopulatedEngine* const engines_LI[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo,
@@ -552,7 +552,7 @@ const PrepopulatedEngine* const engines_LI[] = {
 
 // Lithuania
 const PrepopulatedEngine* const engines_LT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -561,7 +561,7 @@ const PrepopulatedEngine* const engines_LT[] = {
 
 // Luxembourg
 const PrepopulatedEngine* const engines_LU[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo,
@@ -570,7 +570,7 @@ const PrepopulatedEngine* const engines_LU[] = {
 
 // Latvia
 const PrepopulatedEngine* const engines_LV[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yandex_ru,
     &yahoo,
@@ -579,7 +579,7 @@ const PrepopulatedEngine* const engines_LV[] = {
 
 // Libya
 const PrepopulatedEngine* const engines_LY[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_com,
@@ -588,7 +588,7 @@ const PrepopulatedEngine* const engines_LY[] = {
 
 // Morocco
 const PrepopulatedEngine* const engines_MA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_fr,
     &yandex_com,
@@ -597,7 +597,7 @@ const PrepopulatedEngine* const engines_MA[] = {
 
 // Monaco
 const PrepopulatedEngine* const engines_MC[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo_fr,
@@ -606,7 +606,7 @@ const PrepopulatedEngine* const engines_MC[] = {
 
 // Moldova
 const PrepopulatedEngine* const engines_MD[] = {
-    &google,
+    &googleen, &google,
     &yandex_ru,
     &bing,
     &mail_ru,
@@ -615,7 +615,7 @@ const PrepopulatedEngine* const engines_MD[] = {
 
 // Montenegro
 const PrepopulatedEngine* const engines_ME[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -624,7 +624,7 @@ const PrepopulatedEngine* const engines_ME[] = {
 
 // Macedonia
 const PrepopulatedEngine* const engines_MK[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -633,7 +633,7 @@ const PrepopulatedEngine* const engines_MK[] = {
 
 // Mexico
 const PrepopulatedEngine* const engines_MX[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_mx,
     &duckduckgo,
@@ -642,7 +642,7 @@ const PrepopulatedEngine* const engines_MX[] = {
 
 // Malaysia
 const PrepopulatedEngine* const engines_MY[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_my,
     &duckduckgo,
@@ -651,7 +651,7 @@ const PrepopulatedEngine* const engines_MY[] = {
 
 // Nicaragua
 const PrepopulatedEngine* const engines_NI[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -660,7 +660,7 @@ const PrepopulatedEngine* const engines_NI[] = {
 
 // Netherlands
 const PrepopulatedEngine* const engines_NL[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_nl,
     &duckduckgo,
@@ -669,7 +669,7 @@ const PrepopulatedEngine* const engines_NL[] = {
 
 // Norway
 const PrepopulatedEngine* const engines_NO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -678,7 +678,7 @@ const PrepopulatedEngine* const engines_NO[] = {
 
 // New Zealand
 const PrepopulatedEngine* const engines_NZ[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_nz,
     &duckduckgo,
@@ -687,7 +687,7 @@ const PrepopulatedEngine* const engines_NZ[] = {
 
 // Oman
 const PrepopulatedEngine* const engines_OM[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -696,7 +696,7 @@ const PrepopulatedEngine* const engines_OM[] = {
 
 // Panama
 const PrepopulatedEngine* const engines_PA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -705,7 +705,7 @@ const PrepopulatedEngine* const engines_PA[] = {
 
 // Peru
 const PrepopulatedEngine* const engines_PE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_pe,
     &ecosia,
@@ -714,7 +714,7 @@ const PrepopulatedEngine* const engines_PE[] = {
 
 // Philippines
 const PrepopulatedEngine* const engines_PH[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &ecosia,
@@ -723,7 +723,7 @@ const PrepopulatedEngine* const engines_PH[] = {
 
 // Pakistan
 const PrepopulatedEngine* const engines_PK[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -732,7 +732,7 @@ const PrepopulatedEngine* const engines_PK[] = {
 
 // Poland
 const PrepopulatedEngine* const engines_PL[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -741,7 +741,7 @@ const PrepopulatedEngine* const engines_PL[] = {
 
 // Puerto Rico
 const PrepopulatedEngine* const engines_PR[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -750,7 +750,7 @@ const PrepopulatedEngine* const engines_PR[] = {
 
 // Portugal
 const PrepopulatedEngine* const engines_PT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -759,7 +759,7 @@ const PrepopulatedEngine* const engines_PT[] = {
 
 // Paraguay
 const PrepopulatedEngine* const engines_PY[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -768,7 +768,7 @@ const PrepopulatedEngine* const engines_PY[] = {
 
 // Qatar
 const PrepopulatedEngine* const engines_QA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -777,7 +777,7 @@ const PrepopulatedEngine* const engines_QA[] = {
 
 // Romania
 const PrepopulatedEngine* const engines_RO[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -786,7 +786,7 @@ const PrepopulatedEngine* const engines_RO[] = {
 
 // Serbia
 const PrepopulatedEngine* const engines_RS[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -795,7 +795,7 @@ const PrepopulatedEngine* const engines_RS[] = {
 
 // Russia
 const PrepopulatedEngine* const engines_RU[] = {
-    &google,
+    &googleen, &google,
     &yandex_ru,
     &mail_ru,
     &bing,
@@ -804,7 +804,7 @@ const PrepopulatedEngine* const engines_RU[] = {
 
 // Rwanda
 const PrepopulatedEngine* const engines_RW[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -813,7 +813,7 @@ const PrepopulatedEngine* const engines_RW[] = {
 
 // Saudi Arabia
 const PrepopulatedEngine* const engines_SA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_com,
@@ -822,7 +822,7 @@ const PrepopulatedEngine* const engines_SA[] = {
 
 // Sweden
 const PrepopulatedEngine* const engines_SE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_se,
     &duckduckgo,
@@ -831,7 +831,7 @@ const PrepopulatedEngine* const engines_SE[] = {
 
 // Singapore
 const PrepopulatedEngine* const engines_SG[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_sg,
     &duckduckgo,
@@ -840,7 +840,7 @@ const PrepopulatedEngine* const engines_SG[] = {
 
 // Slovenia
 const PrepopulatedEngine* const engines_SI[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo,
@@ -849,7 +849,7 @@ const PrepopulatedEngine* const engines_SI[] = {
 
 // Slovakia
 const PrepopulatedEngine* const engines_SK[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &duckduckgo,
     &yahoo,
@@ -858,7 +858,7 @@ const PrepopulatedEngine* const engines_SK[] = {
 
 // El Salvador
 const PrepopulatedEngine* const engines_SV[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -867,7 +867,7 @@ const PrepopulatedEngine* const engines_SV[] = {
 
 // Syria
 const PrepopulatedEngine* const engines_SY[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_com,
@@ -876,7 +876,7 @@ const PrepopulatedEngine* const engines_SY[] = {
 
 // Thailand
 const PrepopulatedEngine* const engines_TH[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_th,
     &yandex_ru,
@@ -885,7 +885,7 @@ const PrepopulatedEngine* const engines_TH[] = {
 
 // Tunisia
 const PrepopulatedEngine* const engines_TN[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_fr,
     &duckduckgo,
@@ -894,7 +894,7 @@ const PrepopulatedEngine* const engines_TN[] = {
 
 // Turkey
 const PrepopulatedEngine* const engines_TR[] = {
-    &google,
+    &googleen, &google,
     &yandex_tr,
     &yahoo_tr,
     &bing,
@@ -903,7 +903,7 @@ const PrepopulatedEngine* const engines_TR[] = {
 
 // Trinidad and Tobago
 const PrepopulatedEngine* const engines_TT[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -912,7 +912,7 @@ const PrepopulatedEngine* const engines_TT[] = {
 
 // Taiwan
 const PrepopulatedEngine* const engines_TW[] = {
-    &google,
+    &googleen, &google,
     &yahoo_tw,
     &bing,
     &baidu,
@@ -921,7 +921,7 @@ const PrepopulatedEngine* const engines_TW[] = {
 
 // Tanzania
 const PrepopulatedEngine* const engines_TZ[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -930,7 +930,7 @@ const PrepopulatedEngine* const engines_TZ[] = {
 
 // Ukraine
 const PrepopulatedEngine* const engines_UA[] = {
-    &google,
+    &googleen, &google,
     &yandex_ua,
     &bing,
     &duckduckgo,
@@ -939,7 +939,7 @@ const PrepopulatedEngine* const engines_UA[] = {
 
 // United States
 const PrepopulatedEngine* const engines_US[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -948,7 +948,7 @@ const PrepopulatedEngine* const engines_US[] = {
 
 // Uruguay
 const PrepopulatedEngine* const engines_UY[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -957,7 +957,7 @@ const PrepopulatedEngine* const engines_UY[] = {
 
 // Venezuela
 const PrepopulatedEngine* const engines_VE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo_es,
     &duckduckgo,
@@ -966,7 +966,7 @@ const PrepopulatedEngine* const engines_VE[] = {
 
 // Vietnam
 const PrepopulatedEngine* const engines_VN[] = {
-    &google,
+    &googleen, &google,
     &coccoc,
     &yahoo,
     &bing,
@@ -975,7 +975,7 @@ const PrepopulatedEngine* const engines_VN[] = {
 
 // Yemen
 const PrepopulatedEngine* const engines_YE[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &yandex_com,
@@ -984,7 +984,7 @@ const PrepopulatedEngine* const engines_YE[] = {
 
 // South Africa
 const PrepopulatedEngine* const engines_ZA[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -993,7 +993,7 @@ const PrepopulatedEngine* const engines_ZA[] = {
 
 // Zimbabwe
 const PrepopulatedEngine* const engines_ZW[] = {
-    &google,
+    &googleen, &google,
     &bing,
     &yahoo,
     &duckduckgo,
--
2.25.1
